src/statistiek_hub/resources/temporal_dimension_resource.py

```diff
@@ -0,0 +1,41 @@
+from django.core.exceptions import ValidationError
+from import_export.fields import Field
+from import_export.resources import ModelResource
+from import_export.widgets import ForeignKeyWidget
+
+from statistiek_hub.utils.datetime import convert_to_datetime
+from statistiek_hub.utils.check_import_fields import check_missing_import_fields
+
+from referentie_tabellen.models import TemporalDimensionType
+from statistiek_hub.models.temporal_dimension import TemporalDimension
+
+
+class TemporalDimensionResource(ModelResource):
+    type = Field(
+        column_name="type",
+        attribute="type",
+        widget=ForeignKeyWidget(TemporalDimensionType, field="name"),
+    )
+
+    def before_import(self, dataset, using_transactions, dry_run, **kwargs):
+
+        # check column_names importfile
+        expected = ["type", "startdate"]
+
+        error = check_missing_import_fields(fields=dataset.headers, expected=expected )
+        if error:
+            # to speed validation -> if errors empty dataset so no row's will be checked
+            del dataset[0 : len(dataset)]
```

Comment:
Waarschijnlijk mis ik iets hier. Voor mijn gevoel zou deze regel niets moeten doen omdat `dataset` niet meer gebruikt zou worden na de `raise` op de volgende regel. Of gebeurt er hierna toch nog wat met `dataset`?

Reply:
Hier is het idd niet nodig. Maar als er bv nog een hook before_import_row() achter aankomt dan zullen eerst alle rijen bekeken worden voordat de error wordt geserveerd. Dit heeft te maken met het opslaan van de errors in result door de module import-export. Dus het is hier overbodig maar laat het toch staan ivm eventuele aanvullingen later.